recipes/jasper/all/conanfile.py

```diff
@@ -25,88 +28,89 @@ class JasperConan(ConanFile):
-            "set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)",
-            "set(CMAKE_INSTALL_RPATH\n		  \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}\")",
-        ]
-        for cmd_to_remove in cmds_to_remove:
-            tools.replace_in_file(cmakelists, cmd_to_remove, "")
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        generator = self.conf.get("tools.cmake.cmaketoolchain:generator", check_type=str, default="Ninja")
+        tc = CMakeToolchain(self, generator=generator)
+        tc.variables["JAS_ENABLE_DOC"] = False
+        tc.variables["JAS_ENABLE_LATEX"] = False
+        tc.variables["JAS_ENABLE_PROGRAMS"] = False
+        tc.variables["JAS_ENABLE_SHARED"] = self.options.shared
+        tc.variables["JAS_LIBJPEG_REQUIRED"] = "REQUIRED"
+        tc.variables["JAS_ENABLE_OPENGL"] = False
+        tc.variables["JAS_ENABLE_LIBJPEG"] = True
+        if cross_building(self):
+            tc.cache_variables["JAS_CROSSCOMPILING"] = True
+            tc.cache_variables["JAS_STDC_VERSION"] = "199901L"
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        env = VirtualBuildEnv(self)
+        env.generate()
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE*", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "COPYRIGHT*", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         if self.settings.os == "Windows":
             for dll_prefix in ["concrt", "msvcp", "vcruntime"]:
-                tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"),
-                                           "{}*.dll".format(dll_prefix))
-        self._create_cmake_module_variables(
+                rm(self, "{}*.dll".format(dll_prefix), os.path.join(self.package_folder, "bin"))
+        self._create_cmake_module_variables(self,
             os.path.join(self.package_folder, self._module_file_rel_path)
         )
 
     @staticmethod
-    def _create_cmake_module_variables(module_file):
+    def _create_cmake_module_variables(conanfile, module_file):
         content = textwrap.dedent("""\
             if(DEFINED Jasper_FOUND)
                 set(JASPER_FOUND ${Jasper_FOUND})
```

Comment:
needs to be fixed for CMakeDeps: https://github.com/conan-io/conan-center-index/issues/12954